.github/workflows/gh-build.yml

```diff
@@ -67,27 +43,47 @@ jobs:
           role-duration-seconds: 28800 # 8 hours
           role-to-assume: arn:aws:iam::279114543810:role/gha-oidc-nv-legate
 
-      - name: Create conda env
-        shell: su coder {0}
-        run: cd ~/; exec entrypoint make-conda-env;
+      - name: Build docker image
+        run: |
+          echo BUILD_TARGET: ${{inputs.build-target}}
+          echo USE_CUDA: ${{env.USE_CUDA}}
+
+          docker build \
+            --build-arg AWS_SESSION_TOKEN=${{env.AWS_SESSION_TOKEN}} \
+            --build-arg AWS_ACCESS_KEY_ID=${{env.AWS_ACCESS_KEY_ID}} \
+            --build-arg AWS_SECRET_ACCESS_KEY=${{env.AWS_SECRET_ACCESS_KEY}} \
+            --build-arg GITHUB_TOKEN=${{env.GITHUB_TOKEN}} \
```

Comment:
This embeds these vars in the history of the docker image. Please mount secrets into the image instead of define them as build args.

Reply:
Since we are using a multistage docker build and the final image is created from stage0 (which does not define or use these variables) and since these variables are only used in the "build" stage (which is discarded and does not go into the final image) these will not be visible in the image history: https://github.com/nv-legate/legate.core/actions/runs/5553609192/jobs/10142367105

Reply:
There is also an open PR to update the version of docker in the base runner image (https://github.com/actions/runner/pull/2694)

Reply:
Marking resolved as this would only show up on the intermediate image.